src/Components/Puzzle/Puzzle.js

```diff
@@ -0,0 +1,78 @@
+import React, { Component } from 'react';
+import './Puzzle.css';
+import rainbow from '../../rainbow.jpg';
+import Tile from '../Tile/Tile.js';
+
+class Puzzle extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loadingPicture: true,
+      puzzleImg: null
+    }
+  }
+
+  createTiles(xSplits, ySplits) {
+    let {width, height} = this.state.puzzleImg;
+    let dx = Math.floor( width / xSplits );
+    let dy = Math.floor( height / ySplits );
+    let xLeftover = width % dx
+    let yLeftover = height % dy
+    let xOffset, yOffset, tiles = [];
+    for (let y = 0; y < ySplits; y++) {
+      for (let x = 0; x < xSplits; x++) {
+        if (xLeftover) {
+          xOffset = x;
+          xLeftover--;
+        }
+        if (yLeftover) {
+          yOffset = y;
+          yLeftover--;
+        }
```

Comment:
Looks like this is for the leftover pixels you had mentioned previously, definitely makes sense to add them onto the tile. Nice how it kinda cleans its self up too, clever.